core/src/test/java/org/elasticsearch/action/admin/indices/mapping/put/PutMappingRequestTests.java

```diff
@@ -64,4 +71,42 @@ public void testBuildFromSimplifiedDef() {
                 () -> PutMappingRequest.buildFromSimplifiedDef("type", "only_field"));
         assertEquals("mapping source must be pairs of fieldnames and properties definition.", e.getMessage());
     }
+
+    public void testPutMappingRequestFromOldVersion() throws IOException {
+        // this is hacky but here goes
+        PutMappingRequest request = new PutMappingRequest("foo");
+        String mapping = YamlXContent.contentBuilder().startObject().field("foo", "bar").endObject().string();
+        request.source(mapping, XContentType.JSON); // THIS IS NOT A BUG! Intentionally specifying the wrong type so we serialize it
```

Comment:
what practical usecase does this test? user is providing the mapping in yaml but?